### PR TITLE
added akeyless k8s auth option

### PR DESCRIPTION
**Pull request by renanaAkeyless**
for akeyless provider authenticate with k8s auth method:
adding support for segregation from the requesting namespace
by adding a new kubernetesAuth configuration in Akeyless provider where the user can define serviceAccountRef
or secretRef for obtaining the correct jwt

### CODE REVIEW

**Comment by renanaAkeyless** [comment]
@moolen how can I rerun the `CLI / lint `task?

**Comment by moolen** [comment]
I clicked the button! :heavy_check_mark:

**Comment by moolen** [comment]
I or someone else need to find some time to review it :)

File: apis/externalsecrets/v1beta1/secretstore_akeyless_types.go [hunk 1]
```diff
@@ -29,7 +29,16 @@ type AkeylessProvider struct {
 }
 
 type AkeylessAuth struct {
+
+	// Reference to a Secret that contains the detailed
```
Comment (moolen) [inline apis/externalsecrets/v1beta1/secretstore_akeyless_types.go:33]:
```suggestion
	// Reference to a Secret that contains the details
```

File: docs/provider/akeyless.md [hunk 2]
```diff
@@ -16,34 +16,49 @@ The supported auth-methods and their params are:
 | `gcp` |      The gcp audience                                                      |
 | `azure_ad` |  azure object id  (optional)                                                          |
 
-form more information about [Akeyless Authentication Methods](https://docs.akeyless.io/docs/access-and-authentication-methods)
+For more information see [Akeyless Authentication Methods](https://docs.akeyless.io/docs/access-and-authentication-methods)
 
-### Akeless credentials secret
+### Creating an Akeyless Ccredentials Ssecret
```
Comment (moolen) [inline docs/provider/akeyless.md:21]:
```suggestion
### Creating an Akeyless Credentials Secret
```

File: pkg/provider/akeyless/akeyless_api.go [hunk 3]
```diff
@@ -240,6 +248,117 @@ func (a *akeylessBase) getCloudID(provider, accTypeParam string) (string, error)
 	return cloudID, err
 }
 
+func (a *akeylessBase) getK8SServiceAccountJWT(ctx context.Context, kubernetesAuth *esv1beta1.AkeylessKubernetesAuth) (string, error) {
+	if kubernetesAuth.ServiceAccountRef != nil {
+		// Kubernetes <v1.24 fetch token via ServiceAccount.Secrets[]
+		jwt, err := a.getJWTFromServiceAccount(ctx, kubernetesAuth.ServiceAccountRef)
```
Comment (moolen) [inline pkg/provider/akeyless/akeyless_api.go:254]: To make that simpler, this `getJWTFromServiceAccount()` (pulling it from secrets) can be omitted completely unless `Akeyless` cares about the `iss` claim. I've removed that approach from the Kubernetes provider as well. AFAIK only vault is (or better was <1.9) sensible about the `iss` claim.

**Review by moolen — COMMENTED** [review]
LGTM :+1: , left a comment about Kubernetes auth approach, maybe this can be simplified. Let me know what you think.

File: apis/externalsecrets/v1beta1/secretstore_akeyless_types.go (hunk 1 above)
Comment (renanaAkeyless) [reply apis/externalsecrets/v1beta1/secretstore_akeyless_types.go]: fixed.. thanks!

File: docs/provider/akeyless.md (hunk 2 above)
Comment (renanaAkeyless) [reply docs/provider/akeyless.md]: fixed.. thanks!

File: pkg/provider/akeyless/akeyless_api.go (hunk 3 above)
Comment (renanaAkeyless) [reply pkg/provider/akeyless/akeyless_api.go:254]: for now, Akeyless performs ISS Claim validation, so for now I prefer to leave it as is

**Review by moolen — APPROVED** [review]

**Pull request merged**